app/assets/javascripts/controllers/container_deployment/deploy-provider-controller.js

```diff
@@ -0,0 +1,254 @@
+      $scope.deployInProgress = false;
+      $scope.nextButtonTitle = __("Next >");
+    };
+
+    var create_auth_object = function () {
+      var auth = {};
+      switch ($scope.data.authentication.mode) {
+        case 'all':
+          auth.type = "AuthenticationAllowAll";
+          break;
+        case 'htPassword':
+          auth.type = 'AuthenticationHtpasswd';
+          auth.users = $scope.data.authentication.htPassword.users;
+          break;
+        case 'ldap':
+          auth.type = 'AuthenticationLdap';
+          auth.id = $scope.data.authentication.ldap.id;
+          auth.email = $scope.data.authentication.ldap.email;
+          auth.name = $scope.data.authentication.ldap.name;
+          auth.username = $scope.data.authentication.ldap.username;
+          auth.bind_dn = $scope.data.authentication.ldap.bindDN;
+          auth.password = $scope.data.authentication.ldap.bindPassword;
+          auth.certificate_authority = $scope.data.authentication.ldap.ca;
+          auth.insecure = $scope.data.authentication.ldap.insecure;
+          auth.url = $scope.data.authentication.ldap.url;
+          break;
+        case 'requestHeader':
+          auth.type = 'AuthenticationRequestHeader';
+          auth.request_header_challenge_url = $scope.data.authentication.requestHeader.challengeUrl;
+          auth.request_header_login_url = $scope.data.authentication.requestHeader.loginUrl;
+          auth.certificate_authority = $scope.data.authentication.requestHeader.clientCA;
+          auth.request_header_headers = $scope.data.authentication.requestHeader.headers;
+          break;
+        case 'openId':
+          auth.type = 'AuthenticationOpenId';
+          auth.userid = $scope.data.authentication.openId.clientId;
+          auth.password = $scope.data.authentication.openId.clientSecret;
+          auth.open_id_sub_claim = $scope.data.authentication.openId.subClaim;
+          auth.open_id_authorization_endpoint = $scope.data.authentication.openId.authEndpoint;
+          auth.open_id_token_endpoint = $scope.data.authentication.openId.tokenEndpoint;
+          break;
+        case 'google':
+          auth.type = 'AuthenticationGoogle';
+          auth.userid = $scope.data.authentication.google.clientId;
+          auth.password = $scope.data.authentication.google.clientSecret;
+          auth.google_hosted_domain = $scope.data.authentication.google.hostedDomain;
+          break;
+        case 'github':
+          auth.type = 'AuthenticationGithub';
+          auth.userid = $scope.data.authentication.github.clientId;
+          auth.password = $scope.data.authentication.github.clientSecret;
+          break;
+      }
+      return auth;
+    };
+
+    var create_nodes_object = function() {
+      var nodes = [];
+      var id = "";
+      var name = "";
+      var publicName = "";
+      $scope.nodeData.allNodes.forEach(function(item) {
+        if ($scope.data.provisionOn == 'existingVms') {
```

Comment:
> Is it intentional that for existingVms you'll use the publicName of the previous node, and for noProvider, you'll reuse the id of the previous node?
> 
> Because if not, var id, name, publicName should really happen only within that forEach..
> 
> (And if it is, it warrants a comment explaining why)

moved inside the forEach, thanks.